### PR TITLE
Quick fix #4716

### LINKED ISSUE #4716: GroundHeatExchangerHorizontalTrench : incomplete IDF translation
Issue overview
--------------
Incomplete IDF translation of [GroundHeatExchangerHorizontalTrench](https://s3.amazonaws.com/openstudio-sdk-documentation/cpp/OpenStudio-3.4.0-doc/model/html/classopenstudio_1_1model_1_1_ground_heat_exchanger_horizontal_trench.html).

## Current Behavior
After having added/linked the object to a model (e.g. Benchmark Warehouse), OpenStudio in turn generates the following IDF file objects:
```
GroundHeatExchanger:HorizontalTrench,
  Ground Heat Exchanger Horizontal Trench 1, !- Name
  Node 1,                                 !- Inlet Node Name
  Node 2,                                 !- Outlet Node Name
  0.004,                                  !- Design Flow Rate {m3/s}
  75,                                     !- Trench Length in Pipe Axial Direction {m}
  2,                                      !- Number of Trenches
  2,                                      !- Horizontal Spacing Between Pipes {m}
  0.016,                                  !- Pipe Inner Diameter {m}
  0.02667,                                !- Pipe Outer Diameter {m}
  1.25,                                   !- Burial Depth {m}
  1.08,                                   !- Soil Thermal Conductivity {W/m-K}
  962,                                    !- Soil Density {kg/m3}
  2576,                                   !- Soil Specific Heat {J/kg-K}
  0.3895,                                 !- Pipe Thermal Conductivity {W/m-K}
  641,                                    !- Pipe Density {kg/m3}
  2405,                                   !- Pipe Specific Heat {J/kg-K}
  30,                                     !- Soil Moisture Content Percent {percent}
  50,                                     !- Soil Moisture Content Percent at Saturation {percent}
  ,                                       !- Undisturbed Ground Temperature Model Type
  Ground Heat Exchanger Horizontal Trench 1 Ground Model, !- Undisturbed Ground Temperature Model Name
  0.408;                                  !- Evapotranspiration Ground Cover Parameter

Site:GroundTemperature:Undisturbed:KusudaAchenbach,
  Ground Heat Exchanger Horizontal Trench 1 Ground Model, !- Name
  ,                                       !- Soil Thermal Conductivity {W/m-K}
  ,                                       !- Soil Density {kg/m3}
  ,                                       !- Soil Specific Heat {J/kg-K}
  15.5,                                   !- Average Soil Surface Temperature {C}
  12.8,                                   !- Average Amplitude of Surface Temperature {deltaC}
  17.3;                                   !- Phase Shift of Minimum Surface Temperature {days}
```
A subsequent EnergyPlus run will fail with the following errors in _eplusout.err_:
```
Program Version,EnergyPlus, Version 22.1.0-ed759b17ee, YMD=2022.10.14 14:11,
   ** Severe  ** <root>[GroundHeatExchanger:HorizontalTrench][Ground Heat Exchanger Horizontal Trench 1] - Missing required property 'undisturbed_ground_temperature_model_type'.
   ** Severe  ** <root>[Site:GroundTemperature:Undisturbed:KusudaAchenbach][Ground Heat Exchanger Horizontal Trench 1 Ground Model] - Missing required property 'soil_density'.
   ** Severe  ** <root>[Site:GroundTemperature:Undisturbed:KusudaAchenbach][Ground Heat Exchanger Horizontal Trench 1 Ground Model] - Missing required property 'soil_specific_heat'.
   ** Severe  ** <root>[Site:GroundTemperature:Undisturbed:KusudaAchenbach][Ground Heat Exchanger Horizontal Trench 1 Ground Model] - Missing required property 'soil_thermal_conductivity'.
   **  Fatal  ** Errors occurred on processing input file. Preceding condition(s) cause termination.
   ...Summary of Errors that led to program termination:
   ..... Reference severe error count=4
   ..... Last severe error=<root>[Site:GroundTemperature:Undisturbed:KusudaAchenbach][Ground Heat Exchanger Horizontal Trench 1 Ground Model] - Missing required property 'soil_thermal_conductivity'.
   ************* Warning:  Node connection errors not checked - most system input has not been read (see previous warning).
   ************* Fatal error -- final processing.  Program exited before simulations began.  See previous error messages.
   ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
   ************* EnergyPlus Terminated--Fatal Error Detected. 0 Warning; 4 Severe Errors; Elapsed Time=00hr 00min  0.10sec
```
Same _missing_ properties as those in the generated IDF objects. Filling in the properties by hand:
```
...
50,                                     !- Soil Moisture Content Percent at Saturation {percent}
Site:GroundTemperature:Undisturbed:KusudaAchenbach,  !- Undisturbed Ground Temperature Model Type
Ground Heat Exchanger Horizontal Trench 1 Ground Model, !- Undisturbed Ground Temperature Model Name
```
```
Site:GroundTemperature:Undisturbed:KusudaAchenbach,
  Ground Heat Exchanger Horizontal Trench 1 Ground Model, !- Name
  1.08,                                   !- Soil Thermal Conductivity {W/m-K}
  962,                                    !- Soil Density {kg/m3}
  2576,                                   !- Soil Specific Heat {J/kg-K}
  15.5,                                   !- Average Soil Surface Temperature {C}
  12.8,                                   !- Average Amplitude of Surface Temperature {deltaC}
  17.3;                                   !- Phase Shift of Minimum Surface Temperature {days}
```
... fixes the issue. Simulation runs successfully. Not suggesting results are useful or valid without further tweaking, but E+ isn't reporting any errors.

## Expected Behavior
The generated IDF object properties should be written out explicitly (as suggested above).

## Steps to Reproduce
Issue has been reproduced on more than 1x model (here, Benchmark Warehouse), e.g.:
```
translator = OpenStudio::OSVersion::VersionTranslator.new
file = File.join(__dir__, "warehouse.osm")
path = OpenStudio::Path.new(file)
model = translator.loadModel(path)
model = model.get

trench = OpenStudio::Model::GroundHeatExchangerHorizontalTrench.new(model)

loop = model.getPlantLoopByName("Main Service Water Loop").get
puts loop.addSupplyBranchForComponent(trench) # returns true

alt_file = File.join(__dir__, "alt_warehouse.osm")
model.save(alt_file, true)
```
Prep .osw workflow, then run openstudio.

## Possible Solution
(see Expected Behavior)

## Details
(NA)

### Environment
- MacOS, PC
- SDK v3.4.0

### Context
It's of course fairly straightforward to either fill in the blanks by hand, or write a modest EnergyPlus measure (not an OpenStudio measure) to finalize the process. It would be much simpler for users if the IDF object parameters were filled out explicitly during translation. It seems a fairly straightforward fix. This could be tagged as an enhancement (not a bug), but since the errors are encountered systematically ...

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Minimal Fix #4716

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

**Comment by jmarrec** [comment]
So before fix tests logically fails: https://ci.openstudio.net/blue/organizations/jenkins/openstudio-incremental-ubuntu-18-04/detail/PR-4733/2/pipeline/68#step-72-log-19

File: src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerHorizontalTrench.cpp [hunk 1]
```diff
@@ -164,8 +164,28 @@ namespace energyplus {
 
       groundModel.setName(modelObject.name().get() + " Ground Model");
 
+      idfObject.setString(GroundHeatExchanger_HorizontalTrenchFields::UndisturbedGroundTemperatureModelType,
+                          "Site:GroundTemperature:Undisturbed:KusudaAchenbach");
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerHorizontalTrench.cpp:168]: Write model type

File: src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerHorizontalTrench.cpp [hunk 2]
```diff
@@ -164,8 +164,28 @@ namespace energyplus {
...
+      {
+        auto value = modelObject.soilDensity();
+        groundModel.setDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilDensity, value);
+      }
+
+      // SoilSpecificHeat
+      {
+        auto value = modelObject.soilSpecificHeat();
+        groundModel.setDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilSpecificHeat, value);
+      }
```
Comment (jmarrec) [inline src/energyplus/ForwardTranslator/ForwardTranslateGroundHeatExchangerHorizontalTrench.cpp:187]: Write soil properties on the Kusuda object

File: src/energyplus/Test/GroundHeatExchangerHorizontalTrench_GTest.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  EXPECT_EQ("Site:GroundTemperature:Undisturbed:KusudaAchenbach", kusuda.iddObject().type().valueName());
+  EXPECT_EQ(0.358, idfObject.getDouble(GroundHeatExchanger_HorizontalTrenchFields::EvapotranspirationGroundCoverParameter).get());
+
+  EXPECT_EQ(1.02, kusuda.getDouble(GroundHeatExchanger_HorizontalTrenchFields::SoilThermalConductivity).get());
+  EXPECT_EQ(970.0, kusuda.getDouble(GroundHeatExchanger_HorizontalTrenchFields::SoilDensity).get());
+  EXPECT_EQ(2430.0, kusuda.getDouble(GroundHeatExchanger_HorizontalTrenchFields::SoilSpecificHeat).get());
+  EXPECT_EQ(16.5, idfObject.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::AverageSoilSurfaceTemperature).get());
+  EXPECT_EQ(13.8, idfObject.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::AverageAmplitudeofSurfaceTemperature).get());
+  EXPECT_EQ(18.3, idfObject.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::PhaseShiftofMinimumSurfaceTemperature).get());
+}
```
Comment (jmarrec) [inline src/energyplus/Test/GroundHeatExchangerHorizontalTrench_GTest.cpp:136]: New full FT Gtest

File: src/energyplus/Test/GroundHeatExchangerHorizontalTrench_GTest.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  const WorkspaceObject kusuda = idfObject.getTarget(GroundHeatExchanger_HorizontalTrenchFields::UndisturbedGroundTemperatureModelName).get();
+  EXPECT_EQ(IddObjectType{IddObjectType::Site_GroundTemperature_Undisturbed_KusudaAchenbach}, kusuda.iddObject().type());
+  EXPECT_EQ(0.358, idfObject.getDouble(GroundHeatExchanger_HorizontalTrenchFields::EvapotranspirationGroundCoverParameter).get());
+
+  EXPECT_EQ(1.02, kusuda.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilThermalConductivity).get());
+  EXPECT_EQ(970.0, kusuda.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilDensity).get());
+  EXPECT_EQ(2430.0, kusuda.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::SoilSpecificHeat).get());
+  EXPECT_EQ(16.5, kusuda.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::AverageSoilSurfaceTemperature).get());
+  EXPECT_EQ(13.8, kusuda.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::AverageAmplitudeofSurfaceTemperature).get());
+  EXPECT_EQ(18.3, kusuda.getDouble(Site_GroundTemperature_Undisturbed_KusudaAchenbachFields::PhaseShiftofMinimumSurfaceTemperature).get());
```
Comment (jmarrec) [inline src/energyplus/Test/GroundHeatExchangerHorizontalTrench_GTest.cpp:135]: New full FT Gtest

**Review by tijcolem — APPROVED** [review]
Looks good to me. The BCL tests are failing which is unrelate to this PR.

**Pull request merged**